### DISCUSSION TITLE
Suggest removing all Plaster templates but SimpleModule

### DISCUSSION THREAD

**Discussion by johlju** [gaelcolas/Sampler#415] [Ideas]
The PR https://github.com/gaelcolas/Sampler/pull/412 suggest a change so that a user can create a minimal module with the required build and testpipe line (by answering No to all questions asked by Plaster). I suggest that the template `SimpleModule` should be the only way to create a module using `New-SampleModule`, removing the other templates. If there is a need to have unique files for the DSC Community we might need keep that template, but would be good if that was not necessary.

Instead we can
* add additional questions to add more samples during first creation
* add more functionality to `Add-Sample` that can extend an already created module.

By having just one template there is easier to document how to create a module and the different aspects of features, also it is easier to document how to use `Add-Sample` to extend a module.

Thoughts?

**Comment by raandree** [comment]
Agreed. For me it was not clear what the differences are between `CompleteSample`, `dsccommunity`, `SimpleModule` and `SimpleModule_NoBuild`. I like the question-based approach which highlights better things are available.

**Reply by gaelcolas** [reply]
Question-based template selections have some limitations and are not maintainable (huge conditions in Plaster template that has everything and its opposite).
Reducing the number of templates available, yes totally we really need 2 I think.
When we moved each "component" (build, github, git...), the idea is that you can start with a "ready made" template, and then you can add "components by component", easier to maintain and to use.
But the point is also to show people how to make the Sampler extensible by creating customized templates that re-use sampler components (templates).
Company Contoso should be able to create a ContosoSampler module that leverages Sampler and can "track" updates without it being a fork.
